src/analytics/lambdas/scan-metadata-workflow/store-metadata-into-ddb.ts

```diff
@@ -0,0 +1,242 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  DynamoDBClient,
+  BatchWriteItemCommand,
+  BatchWriteItemCommandInput,
+} from '@aws-sdk/client-dynamodb';
+import { logger } from '../../../common/powertools';
+import { aws_sdk_client_common_config } from '../../../common/sdk-client-config';
+import { StoreMetadataBody } from '../../private/model';
+import { getRedshiftClient, executeStatementsWithWait, getRedshiftProps, getStatementResult } from '../redshift-data';
+
+const REDSHIFT_DATA_API_ROLE_ARN = process.env.REDSHIFT_DATA_API_ROLE!;
+const REDSHIFT_DATABASE = process.env.REDSHIFT_DATABASE!;
+
+const { ddb_region, ddb_table_name } = parseDynamoDBTableARN(process.env.METADATA_DDB_TABLE_ARN!);
+
+// Create an Amazon service client object.
+const redshiftDataApiClient = getRedshiftClient(REDSHIFT_DATA_API_ROLE_ARN);
+
+export interface StoreMetadataEvent {
+  detail: StoreMetadataBody;
+}
+
+// Create an Amazon DynamoDB service client object.
+const ddbClient = new DynamoDBClient({
+  ...aws_sdk_client_common_config,
+  region: ddb_region,
+});
+
+/**
+ * The lambda function to get event metadata and properties metadata from redshift. And store them into DDB
+ * @param event StoreMetadataEvent.
+ * @returns.
+ */
+export const handler = async (event: StoreMetadataEvent) => {
+  logger.debug('request event:', JSON.stringify(event));
+
+  const metadataItems: any[] = [];
+  const appId = event.detail.appId;
+  try {
+    await handleEventMetadata(appId, metadataItems);
+
+    await handlePropertiesMetadata(appId, metadataItems);
+
+    await batchWriteIntoDDB(metadataItems);
+
+    return {
+      detail: {
+        appId: appId,
+        status: 'SUCCEED',
+        message: 'store metadata into ddb successfully',
+      },
+    };
+  } catch (err) {
+    if (err instanceof Error) {
+      logger.error('Error when query metadata.', err);
+    }
+    return {
+      detail: {
+        appId: appId,
+        status: 'FAILED',
+        message: 'store metadata into ddb failed',
+      },
+    };
+  }
+};
+
+async function handleEventMetadata(appId: string, metadataItems: any[]) {
+
+  const inputSql = `SELECT id, type, prefix, project_id, app_id, event_name, metadata_source, data_volumel_last_day, platform FROM ${appId}.event_metadata;`;
+
+  const response = await queryMetadata(inputSql);
+
+  // Transform data to DynamoDB format
+  const items = response.Records!.map(record => ({
+    PutRequest: {
+      Item: {
```

Comment:
why not directly returning the item object?

Reply:
Refactored the code to fix it.